sdk-actors/src/main/java/io/dapr/actors/runtime/ActorManager.java

```diff
@@ -291,13 +325,18 @@ private Object invokeMethod(AbstractActor actor, Method method, Object request)
             this.runtimeContext.getActorTypeInformation().getName()));
       }
 
-      return actor.onPreActorMethodInternal(context)
-        .then(func.apply(actor))
-        .flatMap(result -> actor.onPostActorMethodInternal(context).thenReturn(result))
+      Mono<Object> result = actor.onPreActorMethodInternal(context)
+        .then((Mono<Object>)func.apply(actor));
```

Comment:
By separating the chain rollback() is no longer called now if there's a failure.  Why change it from what it was previously?

Reply:
Nevermind, I see now.  I still think it should be one chain of .'s though.

Reply:
OK. I was debugging. I will put these together.